tgui/packages/tgui/interfaces/ForceEvent.tsx

```diff
@@ -0,0 +1,177 @@
+import { paginate } from 'common/collections';
+import { useBackend, useLocalState } from '../backend';
+import { Stack, Button, Icon, Input, Section, Tabs } from '../components';
+import { Window } from '../layouts';
+
+const CATEGORY_PAGE_ITEMS = 4;
+const EVENT_PAGE_ITEMS = 2;
+const EVENT_PAGE_MAXCHARS = 48;
+
+/**
+ * Same as paginate, but respecting event names with a character max length
+ * that will also create a new page if created
+ */
+const paginateEvents = (events: Event[], maxPerPage: number): Event[][] => {
+  const pages: Event[][] = [];
+  let page: Event[] = [];
+  // conditions that make a new page
+  let itemsToAdd = maxPerPage;
+  let maxChars = EVENT_PAGE_MAXCHARS;
+
+  for (const event of events) {
+    maxChars -= event.name.length;
+    if (maxChars <= 0) {
+      // would overflow the next line over
+      itemsToAdd = maxPerPage;
+      maxChars = EVENT_PAGE_MAXCHARS - event.name.length;
+      pages.push(page);
+      page = [];
+    }
+    page.push(event);
+    itemsToAdd--;
+    if (!itemsToAdd) {
+      // max amount of items we allow
+      itemsToAdd = maxPerPage;
+      maxChars = EVENT_PAGE_MAXCHARS;
+      pages.push(page);
+      page = [];
+    }
+  }
+  if (page.length) {
+    pages.push(page);
+  }
+  return pages;
+};
+
+type Event = {
+  name: string;
+  description: string;
+  type: string;
+  category: string;
+};
+
+type Category = {
+  name: string;
+  icon: string;
+};
+
+type ForceEventData = {
+  categories: Category[];
+  events: Event[];
+};
+
+export const ForceEvent = (props, context) => {
+  return (
+    <Window theme="admin" title="Force Event" width={450} height={450}>
+      <Window.Content>
+        <Stack vertical fill>
+          <Stack.Item>
+            <EventTabs />
+          </Stack.Item>
+          <Stack.Item grow>
+            <EventSection />
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const PanelOptions = (props, context) => {
+  const [searchQuery, setSearchQuery] = useLocalState(context, 'searchQuery', '');
+
+  const [announce, setAnnounce] = useLocalState(context, 'announce', true);
+
+  return (
+    <Stack width="240px">
+      <Stack.Item>
+        <Icon name="search" />
```

Comment:
this really needs `vertical-align: middle` which will align it to the input box vertically. Alternatively you can use `line-height`.

Reply:
Don't think vertical align functions in our typescript for some reason. Shows up as undefined.

I will try the line-height.

Reply:
Just put an inline tag.

`style={{'vertical-align': 'middle'}}`

Reply:
can also just use a flex instead.
```
<Flex>
  <Flex.Item>
    <Icon name="search" mr={1} />
  </Flex.Item>
  <Flex.Item grow={1}>
    <Input
      placeholder="Search..."
      fluid
      value={searchText}
      onInput={(_, value) => setSearchText(value)}
      onEnter={(_, value) => orbitMostRelevant(value)}
    />
  </Flex.Item>
</Flex>
```

there is also `style={{ 'align-items': 'center' }}`

Reply:
Gonna be honest, i dont know how to use flex